src/EditorFeatures/Test2/GoToHelpers/GoToHelpers.vb

```diff
@@ -40,6 +44,30 @@ Friend Class GoToHelpers
                     Assert.True(actual.CompareTo(expected) = 0,
                                 $"Expected: ({expected}) but got: ({actual})")
                 Next
+
+                Dim actualDefintionsWithoutSpans = context.GetDefinitions().
+                Where(Function(d) d.SourceSpans.IsDefaultOrEmpty).
+                Select(Function(di)
+                           Return String.Format("{0}:{1}",
+                                                String.Join("", di.OriginationParts.Select(Function(t) t.Text)),
+                                                String.Join("", di.NameDisplayParts.Select(Function(t) t.Text)))
+                       End Function).ToList()
+
+                actualDefintionsWithoutSpans.Sort()
+
+                If metadataDefinitions Is Nothing Then
+                    metadataDefinitions = {}
+                End If
+
+                Assert.Equal(actualDefintionsWithoutSpans.Count, metadataDefinitions.Count)
```

Comment:
Just do Assert.Equal -- there's an overload that compares to IEnumerables item by item, and if the count differs will also show you the actual contents which is a lot easier to debug than this.